tket2/src/circuit.rs

```diff
@@ -352,10 +352,14 @@ fn check_hugr(hugr: &impl HugrView, parent: Node) -> Result<(), CircuitError> {
         },
         OpType::DataflowBlock(_) => Ok(()),
         OpType::Case(_) => Ok(()),
-        _ => Err(CircuitError::InvalidParentOp {
-            parent,
-            optype: optype.clone(),
-        }),
+        OpType::TailLoop(_) => Ok(()),
+        _ => {
+            assert_eq!(None, optype.tag().partial_cmp(&OpTag::DataflowParent));
```

Comment:
Avoid panicking in release builds.
```suggestion
            debug_assert_eq!(
                None,
                optype.tag().partial_cmp(&OpTag::DataflowParent),
            );
```